jujugui/static/gui/src/app/components/profile/navigation/navigation.js

```diff
@@ -0,0 +1,62 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+class ProfileNavigation extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeSection: this.props.activeSection || 'models'
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({activeSection: nextProps.activeSection});
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.state.activeSection !== nextProps.activeSection) {
+      return true;
+    }
+    return false;
+  }
+
+  _changeState(label) {
+    this.props.changeState({
+      // We're using a hash for the state for intra profile navigation because
+      // of the ambiguous paths can not easily support yet another collision
+      // case on top of model and entity names.
+      hash: encodeURIComponent(label.toLowerCase())
+    });
+  }
+
+  render() {
+    const links = ['Models', 'Bundles', 'Charms', 'Cloud Credentials']
+      .map(label =>
+        <li
+          className={this.state.activeSection === label.toLowerCase() ? 'active' : ''} // eslint-disable-line max-len
+          key={label}
```

Comment:
I am not sure I like using labels as keys/hashes. Why not building a map? Like
```javascript
const sections = new Map();
sections.set('models', 'Models');
sections.set('charms', 'Charms');
sections.set('bundles', 'Bundles');
sections.set('credentials', 'Cloud Credentials');

const activeSection = this.state.activeSection;
const links = [];
sections.forEach((value, key) => {
  links.push(<li 
    className={key === activeSection ? 'active' : ''}
    key={key}
    onClick={this._changeState.bind(this, key)}>
    {value}
  </li>);
});
```
This way the order is maintained, the map can be expanded later if we need more state, and we can also avoid the arbitrary toLowerCase modification in _changeState.
What do you think?